src/features/game/events/landExpansion/collectComposterProduce.test.ts

```diff
@@ -0,0 +1,176 @@
+import Decimal from "decimal.js-light";
+import { TEST_FARM } from "features/game/lib/constants";
+import { GameState, PlacedItem } from "features/game/types/game";
+import { collectComposterProduce } from "./collectComposterProduce";
+
+const GAME_STATE: GameState = TEST_FARM;
+
+describe("collectComposterProduce", () => {
+  const dateNow = Date.now();
+
+  it("throws an error if building does not exist", () => {
+    expect(() =>
+      collectComposterProduce({
+        state: {
+          ...GAME_STATE,
+          buildings: {},
+        },
+        action: {
+          type: "composterProduce.collected",
+          building: "Basic Composter",
+          buildingId: "123",
+        },
+        createdAt: Date.now(),
+      })
+    ).toThrow("Composter does not exist");
+  });
+
+  it("throws an error if building is not cooking anything", () => {
+    expect(() =>
+      collectComposterProduce({
+        state: {
+          ...GAME_STATE,
+          buildings: {
+            "Basic Composter": [
+              {
+                id: "123",
+                coordinates: { x: 1, y: 1 },
+                createdAt: 0,
+                readyAt: 0,
+              },
+            ],
+          },
+        },
+        action: {
+          type: "composterProduce.collected",
+          building: "Basic Composter",
+          buildingId: "123",
+        },
+        createdAt: Date.now(),
+      })
+    ).toThrow("Composter is not producing anything");
+  });
+
+  it("throws an error if Produce is not ready", () => {
+    expect(() =>
+      collectComposterProduce({
```

Comment:
My understanding is that the input is "produce" (crops) and the output is "fishing items" (earthworm, etc.).

As such, `collectComposterProduce` should probably be renamed -- specifically the `Produce` part.  You aren't creating produce; you're consuming it.